pandas/core/frame.py

```diff
@@ -2590,9 +2590,25 @@ def assign(self, **kwargs):
 
         return data
 
-    def _sanitize_column(self, key, value):
-        # Need to make sure new columns (which go into the BlockManager as new
-        # blocks) are always copied
+    def _sanitize_column(self, key, value, propagate=True):
+        """
+        Ensures new columns (which go into the BlockManager as new blocks) are
+        always copied and converted into an array.
+
+        Parameters
+        ----------
+        key : object
+        value : scalar, Series, or array-like
+        propagate : bool, default True
```

Comment:
no ``broadcast`` is the correct name. I just wanted the explanation to be what you have now.